### DISCUSSION TITLE
Build configurations for different rendering engines

### DISCUSSION THREAD

**Discussion by sonora** [osmandapp/OsmAnd#15635] [Q&A]
Do we, on Jenkins, have a way/script to compile a build which for rendering engine _Version 1_ supports "_fat_" (armeabi-v7a, arm64-v8a, x86, x86_64), and for _Version 2_ just _arm64-v8a_?

I am looking to build a test build I can hand to members of an expedition for practical testing, but do not know their devices, want them to be able to fall back to Version 1 in any case. (I am aware of `assembleAndroidFullOpenglFatDebug`, but that creates a 330MB file apk...)

**Comment by vshcherb** [comment]
Yes, we ```assembleAndroidFullOpenglFatDebug``` (OpenGl) ```assembleAndroidFullLegacyFatDebug``` (smaller no opengl).
We need configuration as well to "exclude " mini basemap probably

**Comment by sonora** [comment]
Yes, but (in terms of that nomenclature) do we have something like `assembleAndroidFullLegacyFatDebug_plus_OpenglArm64Debug `?

Meaning a build which contains **all** Legacy libraries (which can be used as a general fallback), but only the arm64 OpenGL libs (arm 64 suits the majority of modern devices).

**Reply by vshcherb** [reply]
If you build opengl then legacy is always included but unfortunately it's not possible to do what you mention. It's a bit against parallel versions. Probably it can be done as commenting out https://github.com/osmandapp/OsmAnd/blob/master/OsmAnd/build.gradle#L195.  The general approach is to have legacy build and each opengl flavour build

**Comment by sonora** [comment]
Ok, thanks!
